ote_sdk/ote_sdk/utils/dataset_utils.py

```diff
@@ -0,0 +1,191 @@
+            annotation
+            for annotation in gt_item.get_annotations()
+            if not Rectangle.is_full_box(annotation.shape)
+        ]
+        if (
+            not any(label.is_anomalous for label in gt_item.get_shapes_labels())
+            or len(local_annotations) > 0
+        ):
+            local_idx.append(idx)
+    return local_idx
+
+
+def get_local_subset(
+    dataset: DatasetEntity, fully_annotated_idx: Optional[List[int]] = None
+) -> DatasetEntity:
+    """
+    Extract a subset that contains only those dataset items that have local annotations.
+
+    Args:
+        dataset (DatasetEntity): Dataset from which we want to extract the locally annotated subset.
+        fully_annotated_idx (Optional[List[int]]): The indices of the fully annotated dataset items. If not provided,
+            the function will compute the indices before creating the subset.
+
+    Returns:
+        DatasetEntity: Output dataset with only local annotations
+    """
+    local_items = []
+    if fully_annotated_idx is None:
+        fully_annotated_idx = get_fully_annotated_idx(dataset)
+    for idx in fully_annotated_idx:
+        item = dataset[idx]
+
+        local_annotations = [
+            annotation
+            for annotation in item.get_annotations()
+            if not Rectangle.is_full_box(annotation.shape)
+        ]
+        # annotations with the normal label are considered local
+        normal_annotations = [
+            annotation
+            for annotation in item.get_annotations()
+            if not any(label.label.is_anomalous for label in annotation.get_labels())
+        ]
+
+        local_items.append(
+            DatasetItemEntity(
+                media=item.media,
+                annotation_scene=AnnotationSceneEntity(
+                    normal_annotations + local_annotations,
+                    kind=AnnotationSceneKind.ANNOTATION,
+                ),
+                metadata=item.metadata,
+                subset=item.subset,
+                roi=item.roi,
+                ignored_labels=item.ignored_labels,
+            )
+        )
+    return DatasetEntity(local_items, purpose=dataset.purpose)
+
+
+def get_global_subset(dataset: DatasetEntity) -> DatasetEntity:
+    """
+    Extract a subset that contains only the global annotations.
+
+    Args:
+        dataset (DatasetEntity): Dataset from which we want to extract the globally annotated subset.
+
+    Returns:
+        DatasetEntity: Output dataset with only global annotations
+    """
+    global_items = []
+    for item in dataset:
+        global_annotations = [
+            annotation
+            for annotation in item.get_annotations()
+            if Rectangle.is_full_box(annotation.shape)
+        ]
+        global_items.append(
+            DatasetItemEntity(
+                media=item.media,
+                annotation_scene=AnnotationSceneEntity(
+                    global_annotations, kind=AnnotationSceneKind.ANNOTATION
```

Comment:
What about predictions (not ground truth)? I guess, it does not matter for metric evaluation code and these datasets are only created for these purposes. Is it Ok to have `AnnotationSceneKind.ANNOTATION` even for predictions?